syncer/dispatcher.go

```diff
@@ -74,7 +206,7 @@ type SyncRequest struct {
 // height.
 //
 // rawQueue can panic so it shouldn't be used unwrapped
-type rawQueue []*SyncRequest
+type rawQueue []SyncRequest
```

Comment:
Thanks for tip on dropping `index`, I brought it along from example code when getting the heap to work and didn't examine it closely.

I will rename `syncrequest` to `syncwork`.  I'm not getting rid of it entirely because different work items are going to carry different data once we do things right (pubsub message will have whole header, hello will not).